RELEASE.txt

```diff
@@ -11,71 +11,89 @@ make a PR to update these notes after you are done with the release. ;-)
 
-  4. Rename to ``doc/release/release_<major>.<minor>.txt``
+  4. Scan the PR titles for highlights, deprecations, API changes,
+     and bugfixes, and mention these in the relevant sections of the notes.
+     Try to present the information in an expressive way by mentioning
+     the affected functions, elaborating on the changes and their
+     consequences. If possible, organize semantically close PRs in groups.
 
-  5. Copy ``doc/release/release_template.txt`` to
+  5. Rename the file to ``doc/release/release_<major>.<minor>.txt``
+
+  6. Copy ``doc/release/release_template.txt`` to
      ``doc/release/release_dev.txt`` for the next release.
 
-  6. Copy relevant deprecations from ``release_<major>_<minor>.txt``
+  7. Copy relevant deprecations from ``release_<major>_<minor>.txt``
      to ``release_dev.txt``.
 
-- Update the version number in ``skimage/__init__.py`` and commit.
+- Submit the release notes for review by other project maintainers:
+
+  - Create a PR from v<major>.<minor>.x branch to master (at this point,
+    the difference should show the full contents of the release notes).
+  - Discuss with others, and make the changes directly to v<major>.<minor>.x branch.
+  - Once the consensus is found, ask the project maintainers to merge the PR.
+
+- In the master branch, update the version number in ``skimage/__init__.py``
+  to the next ``-dev`` version, commit, and push.
+
+- Back on the release branch, update the version number to stable in
+  ``skimage/__init__.py``, commit, and push.
 
 - Update the docs:
 
   - Edit ``doc/source/_static/docversions.js`` and commit
   - Build a clean version of the docs.  Run ``python setup.py install`` in the
-    root dir, then ``rm -rf build; make html`` in the docs.
+    root dir, then ``rm -rf build; make html`` in the ``doc`` dir.
   - Build using ``make gh-pages``.
   - Update the symlink to ``stable``.
   - Push upstream: ``git push origin gh-pages`` in ``doc/gh-pages``.
 
 - Add the version number as a tag in git::
 
-   git tag -s [-u <key-id>] v<major>.<minor>.0
+   git tag -s -m <github_release_message> [-u <key-id>] v<major>.<minor>.0
 
-  (If you do not have a gpg key, use -m instead; it is important for
-  Debian packaging that the tags are annotated)
+  (If you do not have a GPG key, follow the tutorial to set it up:
+   https://help.github.com/articles/signing-commits-with-gpg/)
 
-- Push the new meta-data to github::
+- Push the new tag to GitHub::
 
-   git push --tags upstream master
+   git push upstream v<major>.<minor>.0
 
   (where ``upstream`` is the name of the
    ``github.com:scikit-image/scikit-image`` repository.)
 
+- Make sure you have all the required write permissions (if not, you will need
+  to ask Stéfan or Juan to grant you the access), specifically, to:
+
+  - https://github.com/scikit-image/scikit-image-wheels
+  - https://pypi.org/project/scikit-image/
+  - https://github.com/scikit-image/scikit-image-web
```

Comment:
Perhaps this mention should be at the very top, so as to not hold up release after all this effort.